subgraphs/aave-forks/src/mapping.ts

```diff
@@ -372,707 +511,590 @@ export function _handleReserveDataUpdated(
+  createLiquidatorPosition: bool = false
 ): void {
-  const market = Market.load(marketId.toHexString());
+  const market = getMarketFromToken(collateralAsset, protocolData);
   if (!market) {
-    log.warning("[Liquidate] Market not found on protocol: {}", [
-      marketId.toHexString(),
+    log.warning("[_handleLiquidate] Market for token {} not found", [
+      collateralAsset.toHexString(),
     ]);
     return;
   }
-  const inputToken = Token.load(market.inputToken);
-  const protocol = getOrCreateLendingProtocol(protocolData);
-
-  // create liquidate entity
-  const id = `${event.transaction.hash.toHexString()}-${event.logIndex.toString()}`;
-  const liquidate = new Liquidate(id);
-
-  // update liquidators account
-  let liquidatorAccount = Account.load(liquidator.toHexString());
-  if (!liquidatorAccount) {
-    liquidatorAccount = createAccount(liquidator.toHexString());
-    liquidatorAccount.save();
-
-    protocol.cumulativeUniqueUsers += 1;
-    protocol.save();
-  }
-  liquidatorAccount.liquidateCount += 1;
-  liquidatorAccount.save();
-  const liquidatorActorID = "liquidator"
-    .concat("-")
-    .concat(liquidator.toHexString());
-  let liquidatorActor = ActorAccount.load(liquidatorActorID);
-  if (!liquidatorActor) {
-    liquidatorActor = new ActorAccount(liquidatorActorID);
-    liquidatorActor.save();
-
-    protocol.cumulativeUniqueLiquidators += 1;
-    protocol.save();
-  }
-
-  // get borrower account
-  let account = Account.load(borrower.toHexString());
-  if (!account) {
-    account = createAccount(borrower.toHexString());
-    account.save();
-
-    protocol.cumulativeUniqueUsers += 1;
-    protocol.save();
+  const manager = new DataManager(
+    market.id,
+    market.inputToken,
+    event,
+    protocolData
+  );
+  const inputToken = manager.getInputToken();
+  let inputTokenPriceUSD = market.inputTokenPriceUSD;
+  if (!inputTokenPriceUSD) {
+    log.warning(
+      "[_handleLiquidate] Price of input token {} is not set, default to 0.0",
+      [inputToken.id.toHexString()]
+    );
+    inputTokenPriceUSD = BIGDECIMAL_ZERO;
   }
-  account.liquidationCount += 1;
-  account.save();
-  const liquidateeActorID = "liquidatee"
-    .concat("-")
-    .concat(borrower.toHexString());
-  let liquidateeActor = ActorAccount.load(liquidateeActorID);
-  if (!liquidateeActor) {
-    liquidateeActor = new ActorAccount(liquidateeActorID);
-    liquidateeActor.save();
-
-    protocol.cumulativeUniqueLiquidatees += 1;
-    protocol.save();
+  const amountUSD = amount
+    .toBigDecimal()
+    .div(exponentToBigDecimal(inputToken.decimals))
+    .times(inputTokenPriceUSD);
+
+  // according to logic in _calculateAvailableCollateralToLiquidate()
+  // liquidatedCollateralAmount = collateralAmount - liquidationProtocolFee
+  // liquidationProtocolFee = bonusCollateral * liquidationProtocolFeePercentage
+  // bonusCollateral = collateralAmount - collateralAmount / liquidationBonus
```

Comment:
typo? collateralAmount - collateralAmount

Reply:
it is correct `collateralAmount - (collateralAmount / liquidationBonus)`. I will reformulate.